### PR TITLE
Allow non admins to request a loan

### PR DESCRIPTION
**Pull request by Callek**

### CODE REVIEW

**Comment by Callek** [comment]
@djmitche @ccooper - What do you think (at a glance)

Few notes:
* I haven't testing the actual submission yet, thats going to be tested tomorrow.
* This is my first use of Angular, and I can easily see this Form/Functionality being extended to merge in the same features that the admin page has with this. (and articulated well) as a future improvement.
* I know this repo is "dead" but my work was already based pre-merge and I figured it would be less confusing to finish it here, and then I'll merge it in myself "upstream" if this passess muster.
* the relengapi<3.0 change is purely to make my travis jobs pass.
* Travis run of this push at https://travis-ci.org/Callek/build-relengapi-slaveloan/builds/66494240

File: relengapi/blueprints/slaveloan/__init__.py [hunk 1]
```diff
@@ -267,7 +273,12 @@ def update_loan_action(action_id, body):
 @bp.route('/')
 @flask_login.login_required
 def root():
-    return render_template('slaveloan_root.html')
+    return angular.template(
+        'slaveloan_root.html',
+        url_for('.static', filename='slaveloan_root.js'),
+        machine_types=api.get_data(get_machine_classes),
+        loanRequestUrl=url_for("slaveloan.new_loan_request"),
```
Comment (djmitche) [inline relengapi/blueprints/slaveloan/__init__.py]: Let's not mix underscores and interCaps here

File: relengapi/blueprints/slaveloan/static/slaveloan_root.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+angular.module('slaveloan', ['relengapi', 'initial_data']);
+
+angular.module('slaveloan').controller('slaveloanController',
+                                    function($scope, restapi, initial_data) {
+    $scope.machineTypes = initial_data.machine_types;
+    $scope.user = initial_data.user.authenticated_email;
+    $scope.loanRequestUrl = initial_data.loanRequestUrl;
+
+    $scope.newLoanModel = {
```
Comment (djmitche) [inline relengapi/blueprints/slaveloan/static/slaveloan_root.js]: This should just be `$scope.newLoan`

File: relengapi/blueprints/slaveloan/static/slaveloan_root.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            data: JSON.stringify({
+                ldap_email: ldap,
+                bugzilla_email: bugzilla,
+                requested_slavetype: slavetype,
+            }),
+            while: 'Submitting Loan Request',
+        }).then(function(response) {
+            if (response.data.result) {
+                var r = response.data.result;
+                alertify("Loan id " + r.id + " successfully filed")
```
Comment (djmitche) [inline relengapi/blueprints/slaveloan/static/slaveloan_root.js]: This should probably clear out the form somehow to prevent immediate re-submission

**Comment by djmitche** [comment]
From the review checklist:

* documentation updated
  * in docstrings and REST API field comments for API changes
    * the docstring for `new_loan_request` needs to be updated to describe the permissions requirements (and more)
* test coverage
  * all new code is 100% covered by tests
    * the new permissions check needs to be tested
    * the root widget template run needs to be tested (just GET the path)
* style (much of this is verified by `validate.sh`
  * all files have an MPL license header
    * no header on `slaveloan_root.js`

**Comment by Callek** [comment]
Re: Review nits not done:

* >Docstring
** Will fix when I do a full slaveloan docstring review (and when I merge "new from admin" with "new request")
* >Test Coverage
** Will fix when I do the full POST round of test coverage, right now this is untested.
** We do have a root path GET tested: def test_ui_root(client):

**Comment by djmitche** [comment]
Testing and documentation have been "later" for quite a while now, and in general those two important bits of software development too often slip into "never" at Mozilla.

I'd really rather not merge any more patches that introduce additional untested, undocumented code.  I won't make you revert the merge, but for the record my comments did not constitute an r+

**Pull request merged**